src/V3Inst.cpp

```diff
@@ -558,15 +548,20 @@ class InstDeVisitor final : public VNVisitor {
                         AstAssign* const assignp = new AstAssign(nodep->fileline(), lhsp, rhsp);
                         nodep->addNextHere(assignp);
                     }
-                    VL_DO_DANGLING(nodep->unlinkFrBack()->deleteTree(), nodep);
+                    VL_DO_DANGLING(pushDeletep(nodep->unlinkFrBack()), nodep);
                 }
+            } else {
+                iterateChildren(nodep);
             }
```

Comment:
Perhaps OK with code flow but probably cleaner to have the iterate children unconditional at bottom of the function, and add a "return;" on other branches where the node is replaced.